src/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -1329,7 +1357,12 @@ internal static unsafe void NumberToStringFormat(ref ValueStringBuilder sb, Numb
                                 break;
                             ++thousandsSepCtr;
                             if (thousandsSepCtr >= thousandsSepPos.Length)
-                                Array.Resize(ref thousandsSepPos, thousandsSepPos.Length * 2);
+                            {
+                                var newThousandsSepPos = new int[thousandsSepPos.Length * 2];
```

Comment:
Should we get this from / return this to an array pool?

Reply:
We could.  But since it was already allocating and this isn't making it worse, I won't do so in this PR.  Please feel free to do so subsequently :)